### DISCUSSION TITLE
pocketbase.service for Digital Ocean Droplet

### DISCUSSION THREAD

**Discussion by Theonlyhamstertoh** [pocketbase/pocketbase#3907] [Q&A] [answered]
Hi! I'm trying to host pocketbase on digital ocean 

I have my folder structured as 
`/opt/pocketbase_0.19.4_linux_amd64/` which contains `pb_data` `pocketbase` and the rest. 

Inside my ubuntu terminal, I am able to succesfully get the pocketbase server running with:
```
// /opt/pocketbase_0.19.4_linux_amd64 folder:

./pocketbase serve --http="yourdomain.com:80" --https="yourdomain.com:443"

```

I want to setup a pocketbase.service according to the production docs with this installation setup

```
touch /lib/systemd/system/pocketbase.service
nano /lib/systemd/system/pocketbase.service
```

```
[Unit]
Description = pocketbase

[Service]
Type           = simple
User           = root
Group          = root
LimitNOFILE    = 4096
Restart        = always
RestartSec     = 5s
StandardOutput = append:/opt/pocketbase_0.19.4_linux_amd64/logs/access.log
StandardError  = append:/opt/pocketbase_0.19.4_linux_amd64/logs/errors.log
ExecStart      = /opt/pocketbase_0.19.4_linux_amd64/pocketbase serve --http="digitalocean.domain.com:80" --https="digitalocean.domain.com:443"

[Install]
WantedBy = multi-user.target
```
```
systemctl enable pocketbase.service
systemctl start pocketbase
systemctl status pocketbase
```

But when I run the above, I get this 
```
Loaded: loaded (/lib/systemd/system/pocketbase.service; enabled; preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Sat 2023-12-09 11:54:44 UTC; 3s ago
    Process: 16693 ExecStart=/opt/pocketbase_0.19.4_linux_amd64/pocketbase serve --http=digitalocean.domain.com:80 --https=digitalocean.domain.com:443 (code=exited, status=209/STDOUT)
   Main PID: 16693 (code=exited, status=209/STDOUT)
        CPU: 959us
```

What am I doing wrong?

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure what could be the cause of the error. Note that you can either manually start the PocketBase `serve` command or leave the systemd to take care of it, but not both at the same time.

Couple things to note:

- make sure that the `/opt/pocketbase_0.19.4_linux_amd64/logs` directory exist. If not, create it:
  ```
  mkdir /opt/pocketbase_0.19.4_linux_amd64/logs
  ```

- you can skip the `--http` and `--https` flags and just pass the domain name to the `serve` command:
  ```js
  /opt/pocketbase_0.19.4_linux_amd64/pocketbase serve digitalocean.domain.com
  ```
  
If the above doesn't fix it, try to run `journalctl -u pocketbase.service` to check for a more detailed error output.

**Reply by Theonlyhamstertoh** [reply]
That was it! Added the /logs folder solved it and now my pockbase is working. Thank you so much @ganigeorgiev !!
